storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10619,29 +10625,37 @@ void Rdb_background_thread::run()
 
     if (local_stop)
     {
+      // If we're here then that's because condition variable was signaled by
+      // another thread and we're shutting down. Break out the loop to make
+      // sure that shutdown thread can proceed.
+      DBUG_ASSERT(ret == 0);
       break;
     }
 
+    // This path should be taken only when the timer expired.
+    DBUG_ASSERT(ret == ETIMEDOUT);
+
     if (local_save_stats)
     {
       ddl_manager.persist_stats();
     }
 
-    // Flush the WAL if need be but don't do it more frequent
-    // than once per second
     timespec ts;
     clock_gettime(CLOCK_REALTIME, &ts);
-    if (ts.tv_sec - ts_next_sync.tv_sec >= 1)
+
+    // Flush the WAL.
+    if (rdb && rocksdb_background_sync)
```

Comment:
Discussed in person. There are no signals generated from ` rdb_queue_save_stats_request()` so the current solution will still work.